### PR TITLE
Updating s3 and jetty deps

### PR DESCRIPTION
**Pull request by ChGen**
Updating s3 and jetty deps. Reduce potential conflicts over jetty, vulnerabilities and AWS compatibility issues.

### CODE REVIEW

**Comment by ChGen** [comment]
@dgarus , please review.

File: debian-adapter/pom.xml [hunk 1]
```diff
@@ -104,25 +104,13 @@ SOFTWARE.
     <dependency>
       <groupId>com.adobe.testing</groupId>
       <artifactId>s3mock</artifactId>
-      <version>2.4.9</version>
+      <version>3.4.0</version>
```
Comment (dgarus) [inline debian-adapter/pom.xml:107]: We should create a property for `s3mock`'s version in the parent pom

**Review by dgarus — CHANGES_REQUESTED** [review]
please check my comment

File: debian-adapter/pom.xml (hunk 1 above)
Comment (ChGen) [reply debian-adapter/pom.xml:107]: @dgarus , I can make local variable, like for `jettyVersion`. For some reason `jettyVersion` is not inherited from parent pom - without local declaration maven does not proceed. Also IDE Idea 2022 & 2023 with WSL2 makes projects with submodules hardly usable due to several bugs, so I open every module as separate project and need local declaration.

Comment (dgarus) [reply debian-adapter/pom.xml:107]: I don't like an approach when you have to change several places to update one artifact version.
Look at the `guava.version` property as an aexample.

Comment (ChGen) [reply debian-adapter/pom.xml:107]: @dgarus , ok, I'll check my environment, maybe also something broken in it.

**Comment by Tzahif** [comment]
@ChGen, because i raised the issues with S3, [debian](https://github.com/artipie/artipie/issues/1367), [rpm](https://github.com/artipie/artipie/issues/1368) and [docker](https://github.com/artipie/artipie/issues/1369), i can help with testing in my envs (minio & real S3 bucket).
i built this fix branch from source and i still get the same errors.
the only thing that works is rpm over minio (not real s3).
debian and docker still doesn't work with this fix, both minio and real S3 bucket.

**Comment by ChGen** [comment]
@Tzahif , thanks for reporting, I'll continue with https://github.com/artipie/artipie/issues/1367 using minIO configuration you described. And with this changes too.

**Comment by ChGen** [comment]
@dgarus , please review.

**Review by dgarus — APPROVED** [review]

**Comment by ChGen** [comment]
Affected issues:
https://github.com/artipie/artipie/issues/1383
https://github.com/artipie/artipie/issues/1368
https://github.com/artipie/artipie/issues/1367

**Pull request merged**